engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoSource.java

```diff
@@ -0,0 +1,45 @@
+package org.enso.interpreter.runtime.data;
+
+import com.oracle.truffle.api.source.Source;
+import com.oracle.truffle.api.source.SourceSection;
+
+/** Wrapper for exposing sources to Enso. Delegates to original methods with no behavior changes. */
+public class EnsoSource {
+  private final Source source;
```

Comment:
Is there any deeper reason for which we want to have such a wrapper instead of exposing `Source` directly?
Or is this just a facade so that if the API changes in the future, we can avoid having to update the Enso-side?

Reply:
Nah it's a dirty hack, you actually cannot expose source over polyglot, truffle will block you. This should actually be a builtin in object, once we finally get such a concept neatly exposed (which may have to happen soon, seeing as extension methods will be removed).